modules/core/src/lib/layer.js

```diff
@@ -540,6 +542,26 @@ export default class Layer extends Component {
     return count(data);
   }
 
+  // Buffer layout describes how many attribute values are packed for each data object
+  // The default (null) is one value each object.
+  // Some data formats (e.g. paths, polygons) have various length. Their buffer layout
+  //  is in the form of [L0, L1, L2, ...]
+  getBufferLayout(props) {
+    props = props || this.props;
+
+    // First Check if bufferLayout is provided as an explicit value
+    if (props.bufferLayout !== undefined) {
```

Comment:
I would have liked seeing this integrated into other binary data props, rather than an independent top-level prop.